src/modules/ekf2/ekf2_main.cpp

```diff
@@ -924,102 +857,33 @@ void Ekf2::Run()
-				_gps_alttitude_ellipsoid[1] = gps.alt_ellipsoid;
-			}
-		}
-
-		if ((_param_ekf2_gps_mask.get() == 0) && gps1_updated) {
-			// When GPS blending is disabled we always use the first receiver instance
-			_ekf.setGpsData(_gps_state[0].time_usec, _gps_state[0]);
-
-		} else if ((_param_ekf2_gps_mask.get() > 0) && (gps1_updated || gps2_updated)) {
-			// blend dual receivers if available
-
-			// calculate blending weights
-			if (!blend_gps_data()) {
-				// handle case where the blended states cannot be updated
-				if (_gps_state[0].fix_type > _gps_state[1].fix_type) {
-					// GPS 1 has the best fix status so use that
-					_gps_select_index = 0;
-
-				} else if (_gps_state[1].fix_type > _gps_state[0].fix_type) {
-					// GPS 2 has the best fix status so use that
-					_gps_select_index = 1;
-
-				} else if (_gps_select_index == 2) {
-					// use last receiver we received data from
-					if (gps1_updated) {
-						_gps_select_index = 0;
-
-					} else if (gps2_updated) {
-						_gps_select_index = 1;
-					}
-				}
-
-				// Only use selected receiver data if it has been updated
-				_gps_new_output_data = (gps1_updated && _gps_select_index == 0) ||
-						       (gps2_updated && _gps_select_index == 1);
-			}
-
-			if (_gps_new_output_data) {
-				// correct the _gps_state data for steady state offsets and write to _gps_output
-				apply_gps_offsets();
-
-				// calculate a blended output from the offset corrected receiver data
-				if (_gps_select_index == 2) {
-					calc_gps_blend_output();
-				}
-
-				// write selected GPS to EKF
-				_ekf.setGpsData(_gps_output[_gps_select_index].time_usec, _gps_output[_gps_select_index]);
-
-				// log blended solution as a third GPS instance
-				ekf_gps_position_s gps;
-				gps.timestamp = _gps_output[_gps_select_index].time_usec;
-				gps.lat = _gps_output[_gps_select_index].lat;
-				gps.lon = _gps_output[_gps_select_index].lon;
-				gps.alt = _gps_output[_gps_select_index].alt;
-				gps.fix_type = _gps_output[_gps_select_index].fix_type;
-				gps.eph = _gps_output[_gps_select_index].eph;
-				gps.epv = _gps_output[_gps_select_index].epv;
-				gps.s_variance_m_s = _gps_output[_gps_select_index].sacc;
-				gps.vel_m_s = _gps_output[_gps_select_index].vel_m_s;
-				gps.vel_n_m_s = _gps_output[_gps_select_index].vel_ned[0];
-				gps.vel_e_m_s = _gps_output[_gps_select_index].vel_ned[1];
-				gps.vel_d_m_s = _gps_output[_gps_select_index].vel_ned[2];
-				gps.vel_ned_valid = _gps_output[_gps_select_index].vel_ned_valid;
-				gps.satellites_used = _gps_output[_gps_select_index].nsats;
-				gps.heading = _gps_output[_gps_select_index].yaw;
-				gps.heading_offset = _gps_output[_gps_select_index].yaw_offset;
-				gps.selected = _gps_select_index;
-
-				// Publish to the EKF blended GPS topic
-				_blended_gps_pub.publish(gps);
-
-				// clear flag to avoid re-use of the same data
-				_gps_new_output_data = false;
+			if (_vehicle_gps_position_sub.copy(&gps)) {
+				gps_message gps_msg{};
```

Comment:
I would still like to use a function such as `fillGpsMsgWithVehicleGpsPosData` (that you forgot to remove) to reduce the amount of code in this massive function.

Reply:
Thanks, I need to take another look. I believe this PR predates that ekf2 refactor.